pkg/onepassword/client/sdk/sdk.go

```diff
@@ -0,0 +1,97 @@
+package sdk
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/1Password/onepassword-operator/pkg/onepassword/model"
+	sdk "github.com/1password/onepassword-sdk-go"
+)
+
+// Config holds the configuration for the 1Password SDK client.
+type Config struct {
+	ServiceAccountToken string
+	IntegrationName     string
+	IntegrationVersion  string
+}
+
+// SDK is a client for interacting with 1Password using the SDK.
+type SDK struct {
+	client *sdk.Client
+}
+
+func NewClient(config Config) (*SDK, error) {
+	client, err := sdk.NewClient(context.Background(),
+		sdk.WithServiceAccountToken(config.ServiceAccountToken),
+		sdk.WithIntegrationInfo(config.IntegrationName, config.IntegrationVersion),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("1password sdk error: %w", err)
+	}
+
+	return &SDK{
+		client: client,
+	}, nil
+}
+
+func (s *SDK) GetItemByID(vaultID, itemID string) (*model.Item, error) {
+	sdkItem, err := s.client.Items().Get(context.Background(), vaultID, itemID)
+	if err != nil {
+		return nil, fmt.Errorf("1password sdk error: %w", err)
+	}
+
+	var item model.Item
+	item.FromSDKItem(&sdkItem)
+	return &item, nil
+}
+
+func (s *SDK) GetItemsByTitle(vaultID, itemTitle string) ([]model.Item, error) {
```

Comment:
Currently what this function does returns the item overviews for the ones that match the title. What we actually want is to return the items with their details that match the title. Therefore, the item overview is not enough since it doesn't contain the secrets.